src/client/job/job_bus.cpp

```diff
@@ -0,0 +1,72 @@
+#include "job_bus.hpp"
+#include "job_queue.hpp"
+#include "client.hpp"
+#include "logger.hpp"
+
+#include <vector>
+#include <ctime>
+#include <iostream>
+
+bool JobBus::m_exit_loop = false;
+JobBus::b_JobsMap JobBus::m_JobBus_map_bool = {};
+JobBus::s_JobsMap JobBus::m_JobBus_map_string = {};
+JobBus::v_JobsMap JobBus::m_JobBus_map_vector = {};
+
+JobBus::JobBus() {
+    
+    m_JobBus_map_bool.emplace(Job::LIST,                               Client::contact_list);
+    m_JobBus_map_bool.emplace(Job::SEARCH,                             Client::contact_search);
+    m_JobBus_map_bool.emplace(Job::ADD,                                Client::contact_add_user);
+    m_JobBus_map_bool.emplace(Job::REMOVE,                             Client::contact_remove_user);
+    m_JobBus_map_bool.emplace(Job::AVAILABLE,                          Client::contact_available);
+    m_JobBus_map_bool.emplace(Job::SETUSER,                            Client::user_set_username);
+    m_JobBus_map_bool.emplace(Job::SETPASS,                            Client::user_set_password);
+    m_JobBus_map_bool.emplace(Job::CREATE,                             Client::user_register_user);
+    m_JobBus_map_bool.emplace(Job::LOGIN,                              Client::user_login);
+    m_JobBus_map_bool.emplace(Job::LOGGED,                             Client::user_get_logged_in);
+    
+    m_JobBus_map_string.emplace(Job::GETUSER,                          Client::user_get_username);
+    
+    m_JobBus_map_vector.emplace(Job::DISP_CONTACTS,                    Client::contact_get_contacts);
+}
```

Comment:
I think it is better for you to choose   

1. have these members static, initializing them here, and not have a constructor.  Whenever you call the constructor will modify the static members.
2. Not have static members. Make them normal members and initialize them in your constructor.

Why do you have 3 hash maps here?

2.